concourse/scripts/run_job.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -eu
+set -o pipefail
+
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+PIPE='create-bosh-cloudfoundry'
+JOB=$1
+TEMP="/tmp/concourse-run-job.${RANDOM}"
+trap "rm -f $TEMP" ERR
+
+export TARGET_CONCOURSE=deployer
+# shellcheck disable=SC2091
+$("${SCRIPT_DIR}/environment.sh")
+"${SCRIPT_DIR}/fly_sync_and_login.sh"
+FLY="$FLY_CMD -t ${FLY_TARGET}"
+
+# 2 steps prevent fly reading empty input in case step 1 fails
```

Comment:
Could you describe the kind of circumstances the first step would fail? Are they returning the correct exit codes? Does `pipefail` not prevent it from proceeding?

It would be preferable not to use a temporary file, if possible.